pkg/providers/cloudstack/cloudstack.go

```diff
@@ -87,6 +94,34 @@ func (p *cloudstackProvider) PostWorkloadInit(ctx context.Context, cluster *type
 }
 
 func (p *cloudstackProvider) UpdateSecrets(ctx context.Context, cluster *types.Cluster) error {
+	var contents bytes.Buffer
+	err := p.createSecrets(ctx, cluster, &contents)
+	if err != nil {
+		return err
+	}
+
+	err = p.providerKubectlClient.ApplyKubeSpecFromBytes(ctx, cluster, contents.Bytes())
+	if err != nil {
+		return fmt.Errorf("loading secrets object: %v", err)
+	}
+	return nil
+}
+
+func (p *cloudstackProvider) createSecrets(ctx context.Context, cluster *types.Cluster, contents *bytes.Buffer) error {
+	if err := p.providerKubectlClient.GetNamespace(ctx, cluster.KubeconfigFile, constants.EksaSystemNamespace); err != nil {
+		if err := p.providerKubectlClient.CreateNamespace(ctx, cluster.KubeconfigFile, constants.EksaSystemNamespace); err != nil {
```

Comment:
I think I have seen this code in multiple places of the code base. Can just move it to somewhere common? If having this namespace has become a prerequisite to our code, we should just make sure it's always created.